tools/serverpod_cli/lib/src/commands/upgrade.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:io';
+
+import 'package:serverpod_cli/src/logger/logger.dart';
+import 'package:serverpod_cli/src/runner/serverpod_command.dart';
+
+import '../generated/version.dart';
+
+class UpgradeCommand extends ServerpodCommand {
+  @override
+  final name = 'upgrade';
+
+  @override
+  final description = 'Upgrade Serverpod to the latest version.';
+
+  @override
+  void run() async {
+    dynamic error;
+    var success = await log.progress(
+        'Running `dart pub global activate serverpod_cli`...', () async {
+      var process = await Process.run(
```

Comment:
Question: What happens if the user cancels the command after it has been initiated? Will the activate still continue to run in the background or do we halt that process?

Reply:
I deleted `serverpod_cli`, tried to do an upgrade by cancelling it in the middle of the process and finally let it run.
<img width="1244" alt="image" src="https://github.com/serverpod/serverpod/assets/34163765/d81c2ce9-8750-4968-b372-59b972cd6608">

Reply:
I'm not sure that the process to update was started in your canceled call.

Could you do the following test after you have moved the binding of the streams to before awaiting the exit code:
1. deactivate serverpod_cli
2. run `dart serverpod_cli.dart upgrade --verbose`
3. Wait until you see that that "Resolving dependencies" is printed.
4. Cancel the command.
5. Possibly wait a little while if there is a background process running that is downloading the command.
6. Then run `serverpod version`

Then take a print screen of the output and post it here.

I suspect that the CLI will have been updated even if the process was canceled.

Reply:
The result is the expected 😃 
<img width="1254" alt="image" src="https://github.com/serverpod/serverpod/assets/34163765/09f114d5-4b2b-4f79-a041-61ef51e9e19d">

Reply:
Also tested on both mac and Windows and it seems to work as expected 👍